src/data/mv_polynomial/quotients.lean

```diff
@@ -0,0 +1,54 @@
+/-
+Copyright (c) 2020 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Riccardo Brasca
+-/
+
+import data.mv_polynomial.comm_ring
+import ring_theory.ideal.operations
+
+/-!
+# Quotients of polynomial rings in several variables.
+
+We set up the basic theory of quotients of polynomial rings in several variables.
+
+## Main definition
+
+* `ideal.quotient.mk.alg` : given an ideal `I`, the quotient map from `(mv_polynomial σ R)`
+as `R`-algebra homomorphism.
+
+## Main results
+
+* `ideal.quotient.mk.alg_mv_poly_surjective` : `ideal.quotient.mk.alg` is surjective
+* `ideal.quotient.mk.alg_mv_poly_ker` : The kernel of `ideal.quotient.mk.alg` is `I`.
+
+-/
+
+noncomputable theory
+
+namespace mv_polynomial
+
+variables {σ : Type*} {R : Type*} [comm_ring R]
+
+instance {I : ideal (mv_polynomial σ R)} : algebra R (ideal.quotient I) :=
+(ring_hom.to_algebra (ring_hom.comp (ideal.quotient.mk I) C))
```

Comment:
Over in #5703 you define a similar structure but without composing with `C`. Does that mean that the `algebra` that you end up with is not the one you want?

Reply:
The difference is that here I consider any ring `R` and I put an `R`-algebra structure on `R/I`. For mv polynomial this becomes an `R[x_i]`-algebra structure on `R[x_i]/I`. There is no problem in it, but very often (at least in commutative algebra/algebraic geometry) we consider `R[x_i]/I` as an `R`-algebra rather than an `R[x_i]`-algebra (for example speaking about finitely generated algebras and stuff like that). I now think approach is better since one can then use `algebra.comap.algebra` to put an `R`-algebra structure on any `R[x_i]`-algebra, since `R[x_i]` already has an `R`-algebra structure.